### ISSUE TITLE
NZ postcode returning as UNDEFINED

### ISSUE THREAD

**Issue by kevin-argus** [openstreetmap/Nominatim#1711]
Hi,

Certain locations across New Zealand, mainly Auckland, have UNDEFINED postcode.
Sample query:
https://nominatim.openstreetmap.org/reverse?format=json&zoom=30&lat=-36.8574&lon=174.7032

```
{
  "place_id": 63829620,
  "licence": "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
  "osm_type": "node",
  "osm_id": 5415600203,
  "lat": "-36.8573891",
  "lon": "174.7033509",
  "display_name": "399, Point Chevalier Road, Point Chevalier, Albert-Eden, Albert-Eden-Puketāpapa Ward, Auckland, UNDEFINED, New Zealand",
  "address": {
    "house_number": "399",
    "road": "Point Chevalier Road",
    "suburb": "Point Chevalier",
    "city": "Albert-Eden",
    "county": "Albert-Eden-Puketāpapa Ward",
    "state": "Auckland",
    "postcode": "UNDEFINED",
    "country": "New Zealand",
    "country_code": "nz"
  },
  "boundingbox": [
    "-36.8574891",
    "-36.8572891",
    "174.7032509",
    "174.7034509"
  ]
}
```

The problem is our solution implements the display_name field which also contains UNDEFINED where the postcode belongs.

The postcode on that specific example is 1022.

Thank you very much.

**Comment by kevin-argus** [comment]
[faulty_postcode_coordinates_list.zip](https://github.com/openstreetmap/Nominatim/files/4315140/faulty_postcode_coordinates_list.zip)
Adding list of more cases that we encounter.
Thank you

**Comment by mtmail** [comment]
That's a data issue in OSM data. Usually we have a little guide to identify the building or street with the wrong postcode 
http://nominatim.org/release-docs/latest/develop/Postcodes/#finding-places-that-share-a-specific-postcode
but the Overpass API doesn't seem to find it.

I found a street with `postal_code=undefined` (Nominatim uppercases postcodes) with other tools and corrected it. https://www.openstreetmap.org/way/7638804

Nominatim updates from OSM data continuously, usually within minutes. For postcodes there is nightly update task though, it's around 2am UTC time and I can't tell if it already ran this night. Might need to wait 24h.

**Comment by mtmail** [comment]
The data change didn't work yet.

In my database I see the place got updated from OSM data and no longer contains the postcode. `select * from place where osm_type='W' and osm_id=7638804;` but after indexing the postcode still exists. `select * from placex where osm_type='W' and osm_id=7638804;` shows `postcode UNDEFINED`. So the Nominatim update logic doesn't remove the postcode correctly.

I changed the OSM data to postcode 1022, maybe overwriting works. Will need to wait 24 again.
